### PR TITLE
Weekly patch release v1.6.5

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Patch Release v1.6.5

```bash
gh pr list  -S "merged:2022-06-01T18:30:00.000Z..2022-07-01T18:30:00.000Z is:merged" --json mergedAt,milestone,url,mergeCommit,title --jq 'sort_by(.mergedAt) | reverse | .[] | select((.milestone.title == "pl:1.6.x")) | [.mergedAt, .url, .mergeCommit.oid, .title] | join(" ")' --limit 100
```

```console
2022-06-29T22:42:34Z https://github.com/Lightning-AI/lightning/pull/13417 1f85b6d6a4b7932fb4b0ff67b3d2fa493ce14ad6 Fix validation when accelerator is a string
2022-06-29T19:09:35Z https://github.com/Lightning-AI/lightning/pull/13396 2dd332f9c795aa5e590dce4d83e76d791a7b43df Call `set_epoch` for distributed batch samplers
2022-06-29T18:44:46Z https://github.com/Lightning-AI/lightning/pull/13259 55e347ba609df1a1248ef1243ff8cbc09fb97bc8 Unpin `protobuf` version and update `tensorboard` version
2022-06-29T17:34:24Z https://github.com/Lightning-AI/lightning/pull/13420 b332b6632821e3f8fd451bbdf158bc9389eea51a Modified python version check to accommodate for legacy version styles
2022-06-29T13:35:57Z https://github.com/Lightning-AI/lightning/pull/13441 b137ef71345a64e16f397ae601caf714a4cb2c54 CI: fix requirements freeze
2022-06-23T14:21:56Z https://github.com/Lightning-AI/lightning/pull/13384 7a3509decb24113f1a563ece0217afe41ef3e4ce Fix mypy errors for model summary utilities
2022-06-21T22:09:12Z https://github.com/Lightning-AI/lightning/pull/13275 bbc51d16a3fef61059a5dc26feef98b8cd8fe72d Update torchmetrics requirement from <=0.7.2,>=0.4.1 to >=0.4.1,<0.9.2 in /requirements
2022-06-21T19:35:34Z https://github.com/Lightning-AI/lightning/pull/13352 e703917f391dab137ac1b3ed61dd991c8bc16bb9 CI: Build new HPU docker image
2022-06-21T16:48:28Z https://github.com/Lightning-AI/lightning/pull/13350 89e2e69b010b862dc39fff7bd78b8ae6b293f183 [BUG] `estimated_stepping_batches` requires distributed comms in `configure_optimizers` for `DeepSpeedStrategy`
2022-06-21T09:06:49Z https://github.com/Lightning-AI/lightning/pull/13268 55b0635a48489c5162796666073489d1b127c907 Update tutorials
2022-06-21T08:52:06Z https://github.com/Lightning-AI/lightning/pull/13261 d24178ec29689c631ea95dccafdeb7786fddc51a Fix torch.distributed._sharded_tensor DeprecationWarning
2022-06-08T18:36:53Z https://github.com/Lightning-AI/lightning/pull/13250 56cd8833733837a5b484af12b4c7d73ceac56a73 CI: Remove simple test `ci_test-base.yml`
2022-06-02T13:39:14Z https://github.com/Lightning-AI/lightning/pull/13185 3c5a8a833e14c3788fa686cd706cbf2ebd899675 Decouple pulling legacy checkpoints from existing GHA workflows and docker files
2022-06-02T12:54:14Z https://github.com/Lightning-AI/lightning/pull/13136 de4ab1c0273f7caa1cca36eebc235ea283558561 update NGC docker
```

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,251 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 
+## [1.7.0] - 2022-MM-DD
```
Comment (awaelchli) [inline CHANGELOG.md:8]: needs to be updated :)

Comment (rohitgr7) [reply CHANGELOG.md]: done!

**Review by Borda — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,20 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 
+## [1.6.5] - 2022-07-05
+
+### Fixed
+
+- Fixed `estimated_stepping_batches` requiring distributed comms in `configure_optimizers` for the `DeepSpeedStrategy` ([#13350](https://github.com/PyTorchLightning/pytorch-lightning/pull/13350))
+
```
Comment (awaelchli) [inline CHANGELOG.md:13]: shouldn't the spaces be removed?

**Review by awaelchli — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]

**Review by Borda — COMMENTED** [review]
go :rabbit:

**Review by justusschock — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Comment by carmocca** [comment]
The release date format was not correct. Also we had not updated the `__version__` field. GPU CI was green beforehand so we are good to merge.

**Pull request merged**